test/integration/smoke/test_nonstrict_affinity_group.py

```diff
@@ -54,7 +55,11 @@ def setUpClass(cls):
 
         zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
         cls.zone = Zone(zone.__dict__)
-        cls.template = get_template(cls.apiclient, cls.zone.id)
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        if cls.hypervisor.lower() not in ["xenserver"]:
+            cls.template = get_template(cls.apiclient, cls.zone.id)
+        else:
+            cls.template = get_test_template(cls.apiclient, cls.zone.id, cls.hypervisor)
```

Comment:
why not like in test_cluster_drs?

Reply:
only xen requires a template with pv drivers / xen tools installed.